opentelemetry-sdk/src/opentelemetry/sdk/metrics/_internal/aggregation.py

```diff
@@ -349,6 +362,316 @@ def collect(
         return current_point
 
 
+# pylint: disable=protected-access
+class _ExponentialBucketHistogramAggregation(_Aggregation[HistogramPoint]):
+    # _min_max_size and _max_max_size are the smallest and largest values
+    # the max_size parameter may have, respectively.
+
+    # _min_max_size is is the smallest reasonable value which is small enough
+    # to contain the entire normal floating point range at the minimum scale.
+    _min_max_size = 2
+
+    # _max_max_size is an arbitrary limit meant to limit accidental creation of
+    # giant exponential bucket histograms.
+    _max_max_size = 16384
+
+    def __init__(
+        self,
+        attributes: Attributes,
+        start_time_unix_nano: int,
+        # This is the default maximum number of buckets per positive or
+        # negative number range.  The value 160 is specified by OpenTelemetry.
+        # See the derivation here:
+        # https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/sdk.md#exponential-bucket-histogram-aggregation)
+        max_size: int = 160,
+    ):
+        super().__init__(attributes)
+        # max_size is the maximum capacity of the positive and negative
+        # buckets.
+        if max_size < self._min_max_size:
+            raise Exception("size {max_size} is smaller than {min_max_size}")
+
+        if max_size > self._max_max_size:
+            raise Exception("size {max_size} is larter than {max_max_size}")
+
+        self._max_size = max_size
+
+        # _sum is the sum of all the values aggregated by this aggregator.
+        self._sum = 0
+
+        # _count is the count of all calls to aggregate.
+        self._count = 0
+
+        # _zero_count is the count of all the calls to aggregate when the value
+        # to be aggregated is exactly 0.
+        self._zero_count = 0
+
+        # _min is the smallest value aggregated by this aggregator.
+        self._min = inf
+
+        # _max is the smallest value aggregated by this aggregator.
+        self._max = -inf
+
+        # _positive holds the positive values.
+        self._positive = Buckets()
+
+        # _negative holds the negative values by their absolute value.
+        self._negative = Buckets()
+
+        # _mapping corresponds to the current scale, is shared by both the
+        # positive and negative buckets.
+        self._mapping = LogarithmMapping(LogarithmMapping._max_scale)
+
+        self._instrument_temporality = AggregationTemporality.DELTA
+        self._start_time_unix_nano = start_time_unix_nano
+
+    def aggregate(self, measurement: Measurement) -> None:
```

Comment:
```suggestion
    def aggregate(self, measurement: Measurement) -> None:
        # Outline of this method:
        # 1. Maintain the _min, _max, _count, _zero_count, and _sum fields
        # 2. Determine the range (positive or negative), then use the absolute value
        # 3. Compute the (absolute) value's index at the current scale
        # 4. Check whether the value's index falls into [_index_start, _index_end]
        # 5. If not, rescale the data structure with a change of scale
        # 6. Then, if the new index falls outside [_index_start, _index_end], adjust the boundaries of the circular backing array
        # 7. Compute the position of the index in the circular backing array
        # 8. Finally, update the bucket.
```

Reading through this, I wanted for comments. (Possibly they should go in-line with the code.)

Reply:
Nice! added comments.